nntrainer/models/neuralnet.cpp

```diff
@@ -433,6 +444,11 @@ int NeuralNetwork::train(std::vector<std::string> values) {
 int NeuralNetwork::train_run() {
   int status = ML_ERROR_NONE;
 
+  if (!continue_train) {
```

Comment:
IMO this should move to `readModel`.
If the user calls the train function twice manually, then it should start from the previously loaded epoch in the model.

Reply:
I think we have to clarify what `continue_train` means.

There seems several cases can be expected when continue_train is **false**

1. Strong reset: at the start of the `train_run()` we reset everything (weights and others...)
2. Weak reset: calling train_run() twice will continue from where the user has stopped. (but the epoch and arguably iteration(?) is changed)
3. No reset(for the current session): callling train_run() trains from stopped iteration and epoch. 

IMO `2` is what I think turning off continue_train. For the case of `3`, as long as `train_run()` finishes. second `train_run()` is essentially noop. even though continue_train is off. 

Our code states, 
```cpp
  bool continue_train; /**< Continue train from the previous state of optimizer
   and iterations */
```

@jijoongmoon  Could you give some opinion as well?

Reply:
The total training weights are divided into two parts - 
1. Model weights - this represents the weights which are needed when model is being inferred. This will just include weights of the layers and other layer configuration related hyper parameters like stride, padding etc.
2. Training weights - this represents weights to facilitate the training and continue from previously saved training instance. This will include optimizer related weights, epochs, iterations etc.

So, with `continue_train`, training weights must be loaded. Without, `continue_train` training weights must not be loaded.
However, there is another meaning to where the continue_train must be applied to - read or train.
If `continue_train` is applied to `train`, then the training_weights must all be reset at every call to train when Training_weights is called.
If `continue_train` is applied to `read/load`, then training weights must not be reset with every call to train.

That said, these both can be merged with `train` itself. `train` function saves the model and training weights at its end, however, it does not load it at its starting. `train` function must be modified to `read/load` at its start, where `read/load` can utilize the `continue_train` and appropriately do the job.

Reply:
I guess this should be handled separately than just a quick fix as some application explicitly calls `readModel()`. I'll leave this as it is for now and deal from #736.